### ISSUE TITLE
How to force in a definitive way the model to respond only withing the data context

### ISSUE THREAD

**Issue by RicBruD4S** [Azure-Samples/azure-search-openai-demo#697]
### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Expected/desired behavior
> Respond only and always withing the context of the data

Hi, I've the requirement to develop the model to respond ONLY to the data sources provided in Azure Cognitive Seatch but the model sometimes goes out-of-context; for example prompting questions such as "who is the president of the USA?" , the model first replies something like: "I don't know, there is no sources about that" but if I continue asking the same or somehow "forcing" the reply, at the end the model replies (sometimes after 2-3 times, sometimes after 6-10 times asking the same question).
How can I limit the model in a definitive way to answer to in-bound questions and discarding anything else out-of-context?
I was thinking to something like if there is no reference for the answer, discard it anyway, but sometimes it happens that the model gives the wrong citation with a out-of-context answer, which is really problematic and erroneous.

Have you any idea on that? Thank you.

**Comment by elhele** [comment]
Hi, I have the same issue and at the moment also don't know how to solve it. Example is attached. It even finds some source, though it is a strictly  technical documentation and cannot contain any information like this.
<img width="1064" alt="Bildschirmfoto 2023-10-02 um 14 37 17" src="https://github.com/Azure-Samples/azure-search-openai-demo/assets/13983474/66031115-5a56-4534-9605-b363cd943c47">

I tried to adjust the temperature but it didn't help.

**Comment by RicBruD4S** [comment]
> Hi, I have the same issue and at the moment also don't know how to solve it. Example is attached. It even finds some source, though it is a strictly technical documentation and cannot contain any information like this. <img alt="Bildschirmfoto 2023-10-02 um 14 37 17" width="1064" src="https://user-images.githubusercontent.com/13983474/271970679-66031115-5a56-4534-9605-b363cd943c47.png">
> 
> I tried to adjust the temperature and lots of different prompts but it didn't help.

Yes indeed, I've also tried to adjust the system prompt trying to convince the AI to answer only context-related questions, nothing seems to help.

**Comment by pamelafox** [comment]
I just tried asking "What is the capitol of the UK?" on the sample data 12 times, and I could not get it to answer anything other than "I'm sorry, but the information provided does not include the capital of the UK."

Can you clarify if you are using the latest version of the repository?
And are you using your own data or the sample data?
When it provides a citation, is it to your own data? Can you log out what search query yielded results in your data? I don't know if the issue is with the Azure Cognitive Search query or the ChatGPT answering without having data.

Thanks! I'd like to get solid examples of it answering off-script (so that we can improve it) but haven't managed to replicate most of the examples.

**Comment by RicBruD4S** [comment]
Hi, here the answers:

- Yes it is the latest version of the repo
- My own data which is just 2 technical PDFs
- When the model answers with the wrong citation, the tought process and the supporting content are completely wrong, there is (of course) no relationship from the technical content with the question (ie: who is the president of the USA?), here below an example:
![Screenshot 2023-10-03 092737](https://github.com/Azure-Samples/azure-search-openai-demo/assets/132678601/f5d4d793-59ea-4dd4-91a2-7a950456d0b4)

**Comment by pamelafox** [comment]
Thanks for the additional details! I realize the thought process does not include the queries.

Can you log or print out the queries sent to ACS in chatreadretrieveread.py? (the query_text variable, before the call to self.search_client.search)

It'd be interesting to see if the issue is due to ChatGPT generating a bad query, or if it's the same query each time.

**Comment by srbalakr** [comment]
Hi @RicBruD4S  is it a pure vector search or hybrid search ?

**Comment by RicBruD4S** [comment]
> Hi @RicBruD4S is it a pure vector search or hybrid search ?

Pure vector search only

**Comment by pamelafox** [comment]
I'll remove stale label since I did see a similar issue today when asking "what does a PM do?" on an index where the user didn't have access to any sources.
